src/Encoder.cpp

```diff
@@ -57,8 +59,7 @@ namespace cppkin
         return m_buffer->getBufferAsString();
     }
 
-    template <class T>
-    std::string EncoderImpl<EncodingTypes::Thrift>::SpansVectorToString(T& spans) const {
+    string EncoderImpl<EncodingTypes::Thrift>::ToString(const std::vector<EncoderContext::ContextElement>& spans) const {
```

Comment:
Excellent, i really like the fact that we have a more tight interface, the only mean to serialize is by using the context and not the Spans directly, excellent work.